src/s2wasm.h

```diff
@@ -159,16 +159,23 @@ class S2WasmBuilder {
     return cashew::IString(str.c_str(), false);
   }
 
-  int32_t getInt() {
-    int32_t ret = 0;
+  uint32_t getInt() {
```

Comment:
Indeed, I'd like to later change the code to parse out the `-` sign first, and negate if required. Maybe also carry the s/u distinction internally, I'm not sure about that one yet.

For now IIUC it does the right thing, without hitting UB.